acstools/polarization_tools.py

```diff
@@ -1,85 +1,328 @@
+Toolkit for analyzing HST/ACS WFC and HRC polarization data. For more information
+about ACS polarization data analysis, see section 5.3 of the ACS Data Handbook:
 
-CHANGE LOG
-----------
-2020-09-22: Creation. (T. Desjardins)
+https://hst-docs.stsci.edu/acsdhb/chapter-5-acs-data-analysis/5-3-polarimetry
 """
 
-import numpy as np
 import os
+import numpy as np
 import yaml
 
+from astropy import units
 from astropy.table import Table
+from astropy.utils.data import get_pkg_data_filename
+
+
+def calc_stokes(pol0, pol60, pol120, c0=1, c60=1, c120=1):
+    """
+    Calculate the Stokes parameters for ACS observations.
+
+    Parameters
+    ----------
+    pol0 : float
+        Measurement in the POL0 filter. Units: electrons or electrons/second.
+    pol60 : float
+        Measurement in the POL60 filter. Units: electrons or electrons/second.
+    pol120 : float
+        Measurement in the POL120 filter. Units: electrons or electrons/second.
+    c0 : float (Default = 1; optional)
+        Efficiency term for the POL0 filter. This comes from the ACS Data Handbook
+        Table 5.6 in Section 5.3.4.
+    c60 : float (Default = 1; optional)
+        Efficiency term for the POL60 filter. This comes from the ACS Data Handbook
+        Table 5.6 in Section 5.3.4.
+    c120 : float (Default = 1; optional)
+        Efficiency term for the POL120 filter. This comes from the ACS Data Handbook
+        Table 5.6 in Section 5.3.4.
+
+    Returns
+    -------
+    i, q, u : tuple of floats
+        Stokes I, Q, and U parameters.
+    """
+
+    r0 = pol0 * c0
+    r60 = pol60 * c60
+    r120 = pol120 * c120
+
+    i = (2 / 3) * (r0 + r60 + r120)
+    q = (2 / 3) * ((2 * r0) - r60 - r120)
+    u = (2 / np.sqrt(3)) * (r60 - r120)
+
+    return i, q, u
+
+
+def calc_fraction(i, q, u, transmission_correction=1):
+    """
+    Method for determining the fractional polarization.
+
+    Parameters
+    ----------
+    i : float
+        Stokes I parameter.
+    q : float
+        Stokes Q parameter.
+    u : float
+        Stokes U parameter.
+    transmission_correction : float (Default = 1)
+        Correction factor to account for the leak of photons with
+        non-parallel electric field position angles. See Section 5.3.4
+        of the ACS Data Handbook.
+
+    Returns
+    -------
+    pol_fraction : float
+        Polarization fraction.
+    """
+
+    pol_fraction = np.sqrt(q**2 + u**2) / i
+    pol_fraction *= transmission_correction
+
+    return pol_fraction
+
+
+def calc_theta(q, u, detector, pav3):
+    """
+    Calculate the position angle of the electric field vector given Stokes Q and U.
+    For ACS, it is also necessary to supply the position angle of the V3 axis of the
+    telescope (PA_V3 in the primary header) and the name of the ACS detector used.
+
+    Parameters
+    ----------
+    q : float
+        Stokes Q parameter.
+    u : float
+        Stokes U parameter.
+    detector : ['wfc', 'hrc']
```

Comment:
Curly bracket indicate the choices are in a set. That is the convention I learned anyway from SSB days.
```suggestion
    detector : {'wfc', 'hrc'}
```